pkg/api/handlers.go

```diff
@@ -39,6 +43,35 @@ func newHandler(context *clusterd.Context, config *Config) *Handler {
 	}
 }
 
+func (h *Handler) Init() error {
```

Comment:
I would suggest to rename this function to other than `init`. Init in go has a special meaning. https://golang.org/doc/effective_go.html#init. Perhaps name it `registerPrometheus`?